pylint/checkers/classes/class_checker.py

```diff
@@ -904,6 +911,33 @@ def _check_proper_bases(self, node: nodes.ClassDef) -> None:
                         confidence=INFERENCE,
                     )
 
+            if isinstance(ancestor, nodes.ClassDef) and ancestor.is_subtype_of(
+                "enum.Flag"
+            ):
+                previous_values, total, union, overlaps = set(), None, None, []
+                for assign_name in node.nodes_of_class(nodes.AssignName):
+                    if isinstance(assign_name.parent, nodes.Arguments):
+                        continue  # Ignore function arguments
+
+                    assigned = assign_name.parent.value
+                    if not isinstance(assigned, nodes.Const):
+                        continue  # Ignore non-literal assignments
+
+                    assert assigned.value is not None
+                    if assigned.value in previous_values:
+                        continue  # Ignore aliases
+                    previous_values.add(assigned.value)
+
+                    # Detect divergence between the sum-of-values and union-of-values
+                    total = assigned.value if total is None else total + assigned.value
```

Comment:
Note: there's an assumption here that subclasses of `enum.Flag` will implement addition/sum operations - I don't know whether that's reasonable (the check could be adjusted to only inspect subclasses of `enum.IntFlag`).